jakelib/deploy.jake

```diff
@@ -1,138 +1,105 @@
-      cmds: [
-        'npm run build-{{cluster_name}}',
-        'docker build -t {{cluster_name}}-{{app_name}} . --no-cache',
-        'docker tag {{cluster_name}}-{{app_name}}:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
-    },
-    'inventory-worker': {
-      cmds: [
-        'docker build -t {{cluster_name}}-{{app_name}} -f docker/worker . --no-cache',
-        'docker tag {{cluster_name}}-{{app_name}}:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
-    },
-    'analytics-mosql-logevents': {
-      cmds: [
-        'docker build -t greenchef/mosql-base -f ./docker/mosql-base/Dockerfile . --no-cache',
-        'docker build -t mosql-logevents -f docker/mosql-logevents/Dockerfile . --no-cache',
-        'docker tag mosql-logevents:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
-    },
-    'analytics-mosql-models': {
-      cmds: [
-        'docker build -t greenchef/mosql-base -f ./docker/mosql-base/Dockerfile . --no-cache',
-        'docker build -t mosql-models -f docker/mosql-models/Dockerfile . --no-cache',
-        'docker tag mosql-models:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
-    },
-    'analytics-mosql-shipping': {
-      cmds: [
-        'docker build -t greenchef/mosql-base -f ./docker/mosql-base/Dockerfile . --no-cache',
-        'docker build -t mosql-shipping -f docker/mosql-shipping/Dockerfile . --no-cache',
-        'docker tag mosql-shipping:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
-    },
-    'jsreports': {
-      cmds: [
-        'docker build -t {{cluster_name}}-{{app_name}} -f Dockerfile . --no-cache',
-        'docker tag {{cluster_name}}-{{app_name}}:latest 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-        'docker push 052248958630.dkr.ecr.us-west-2.amazonaws.com/{{cluster_name}}-{{app_name}}:latest',
-      ]
     }
-  };
+  });
+  return value;
+}
+
+const buildCmdString = (path, cluster_name, app_name) => {
+  const vars = { cluster_name, app_name };
+  const cmds = [
+    `cd ${path}`,
+    ...apps[app_name].cmds,
+  ];
+  return replacer(cmds.join(' && '), vars);
+};
+
+const normalAppDeployCommands = [
+  'docker build -t {{cluster_name}}-{{app_name}} . --no-cache',
```

Comment:
now these vars accessible inside the function... why not do normal interpolation and get rid of the replacer function?

Reply:
ooo, is this the replacer function? I thought it was some special jake thing. I'll look into it.

Reply:
yeah this is the handwound interpolation

Reply:
I'm gonna merge this in and look at replacing the replacer in a different PR.

Reply:
🆗 🙆‍♂ 👌 🔉 💯 🏷 🥇